anoncreds/src/data_types/anoncreds/cred_def.rs

```diff
@@ -1,17 +1,19 @@
-use crate::data_types::identifiers::cred_def::CredentialDefinitionId;
-use crate::data_types::identifiers::schema::SchemaId;
-use crate::data_types::{ConversionError, Validatable, ValidationError};
+use crate::{data_types::ConversionError, impl_anoncreds_object_identifier};
+
+use super::schema::SchemaId;
 
 pub const CL_SIGNATURE_TYPE: &str = "CL";
 
+impl_anoncreds_object_identifier!(CredentialDefinitionId);
+
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub enum SignatureType {
     CL,
 }
 
 impl SignatureType {
-    pub fn from_str(value: &str) -> Result<Self, ConversionError> {
-        match value {
+    pub fn from_string(value: impl Into<String>) -> Result<Self, ConversionError> {
```

Comment:
It might be more idiomatic to implement the FromStr trait instead of an inherent method. This change allocates a new String unnecessarily if an `&str` is passed in.

Reply:
yeah the reason that I did not do that is that from_str does not return a Result. TryFrom would be good here as I assume it does return a result.

Reply:
FromStr::from_str does result a Result, like TryFrom

Reply:
Oh yeah, huh. I tried to fix this issue with implementing it on FromStr, but I think I did From<str>. Do you have a preference for TryFrom or FromStr?

Reply:
I guess my preference is FromStr, although it does require an include when used.